src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -2537,6 +2537,43 @@ def list_resource_links(cmd, scope=None, filter_string=None):
 # endregion
 
 
+# region tags
+def get_tag_at_scope(cmd, resource_id=None):
+    rcf = _resource_client_factory(cmd.cli_ctx)
+    if resource_id is not None:
+        return rcf.tags.get_at_scope(scope=resource_id)
+
+    return rcf.tags.list()
+
+
+def create_or_update_tag_at_scope(cmd, resource_id=None, tags=None, tag_name=None):
+    rcf = _resource_client_factory(cmd.cli_ctx)
+    if resource_id is not None:
+        if tags is None or bool(tags) is False:
+            raise IncorrectUsageError("Tags could not be empty.")
+        Tags = cmd.get_models('Tags')
+        tag_obj = Tags(tags=tags)
+        return rcf.tags.create_or_update_at_scope(scope=resource_id, properties=tag_obj)
+
+    return rcf.tags.create_or_update(tag_name=tag_name)
```

Comment:
> rcf.tags.create_or_update(tag_name=tag_name)

If the `resource_id` is None, can only create `tag_name`, not support the creation of `tag_value`?

Reply:
Yes, this is an existing api: update `tag_value` would be handled in `g.command('add-value', 'create_or_update_value')`.
For the sdk: 
```python
def create_or_update(self, tag_name, custom_headers=None, raw=False, **operation_config):
create_or_update.metadata = {'url': '/subscriptions/{subscriptionId}/tagNames/{tagName}'}
```

Reply:
Got it, thanks~